### PR TITLE
enhancement: add update username method in UserRepository

### PR DESCRIPTION
**Pull request by MarkPotato777**
#### What type of PR is this?

module-iam

#### What this PR does / why we need it:
add `updateUsernameById` method in `UserRepository`

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/metadb/iam/UserRepository.java [hunk 1]
```diff
@@ -98,6 +98,11 @@ List<UserEntity> findByUserIdsOrRoleIds(@Param("userIds") Collection<Long> userI
             nativeQuery = true)
     int updatePassword(@Param("user") UserEntity user);
 
+    @Modifying
+    @Transactional
+    @Query(value = "update iam_user u set u.name=:name Where u.id=:id", nativeQuery = true)
+    int updateUsernameById(@Param("id") Long id, @Param("name") String name);
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/metadb/iam/UserRepository.java:104]: Why add this method without any invoke?

Comment (MarkPotato777) [reply server/odc-service/src/main/java/com/oceanbase/odc/metadb/iam/UserRepository.java:104]: odc-cloud repository needs this method.

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**